locations/spiders/shopee_xpress_tw.py

```diff
@@ -0,0 +1,56 @@
+import json
+from typing import Any
+
+from scrapy import Spider
+from scrapy.http import Response
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+
+SHOPEE_XPRESS = {"brand_wikidata": "Q109676747"}
+SIMPLE_MART = {"brand_wikidata": "Q15914017"}
+
+
+class ShopeeXpressTWSpider(Spider):
+    name = "shopee_xpress_tw"
+    start_urls = [
+        "https://spx.tw/api/service-point/point/around/list?radius=400000000&latitude=25.740539&longitude=121.8713269",
+        "https://spx.tw/api/service-point/point/around/list?radius=400000000&latitude=21.749306&longitude=120.7836804",
+    ]
+
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        for location in response.json()["data"]["list"]:
+            item = DictParser.parse(location)
+            item["name"] = location["alias"]
+            item["website"] = "https://spx.tw/service-point/detail/{}".format(item["ref"])
+
+            if location["sp_point_type"] == 1:
+                item.update(SHOPEE_XPRESS)
+                # Some are "parcel lockers"
+                # others have staff and some ability to buy some things off the shelf
+                # TODO
```

Comment:
This is my blocker. I can't figure it out in the data, but I'm assuming it is there.

Reply:
NSI has them as `shop=convenience` and `amenity=parcel_locker`, it doesn’t feel right, but I don't know enough to argue it.

Reply:
TL;DR

- Unmanned (interact with kiosk) → `amenity=parcel_locker`
![](https://ibw.bwnet.com.tw/ac_gallery/2023/10/3578a8ad-0a03-c0f1-cf92-68b726a8c0f4_620.jpg)
- Manned (interact with staff) → ~`shop=convenience`~ `office=courrier` (as per iigmir)
![](https://ibw.bwnet.com.tw/ac_gallery/2021/11/d1f04fdb-98ab-7770-dd43-10af6d7b2560_620.jpg)

For reference:
https://github.com/osmlab/name-suggestion-index/issues/10890

Reply:
P.S. Also, not thinking the crawl syntax is accurate. 
Shopee might have parcel lockers in some SimpleMart, but not all stores.